### ISSUE TITLE
Why dont we have validation data during cifar trainig?

### ISSUE THREAD

**Issue by isalirezag** [kuangliu/pytorch-cifar#49]
Please correct me if im wrong, but should not we have validation dataset while we train ?
I am looking at cifar examples and i noticed that we dont have validation dataset, we just have train and test.
Should we skip validation? why?
Can you please explain it to me

Thanks

**Comment by amrit110** [comment]
One should always use a validation set if you are tuning stuff, and also to notice overfitting. So, you can take a small set out of the training say 10k out of the 50k and use it as validation. Once you are done tuning, you can train on the whole set again. To be extra rigorous, cross-validation on multiple validation sets shuffled for every experiment is recommended. There has been concerns expressed about this, https://arxiv.org/abs/1806.00451.

**Comment by phelps-matthew** [comment]
I agree with @amrit110 . If you run the training as is, it will inadvertently be tuning itself directly to the test set and achieve overly high accuracies (I was getting over 95% on ResNet 34). 

Here is a sample dataloader than includes the validation. [https://github.com/phelps-matthew/FeatherMap/blob/train/feathermap/data_loader.py](https://github.com/phelps-matthew/FeatherMap/blob/train/feathermap/data_loader.py ).

Cross validation is good here, but can be rather expensive. I am now at least getting consistent accuracies using a validation set of size 10%.
